src/App.tsx

```diff
@@ -38,58 +40,120 @@ export function getReorderedGoods(
   // eslint-disable-next-line no-console
   console.log(sortType, isReversed);
 
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType
+};
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  sortByAlphabet = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  reset = () => {
+    this.setState({ isReversed: false, sortType: SortType.NONE });
```

Comment:
```suggestion
    this.setState({
      isReversed: false,
      sortType: SortType.NONE,
    });
```

Reply:
Writing each property of an object on a new line is a style convention in JavaScript that makes the code easier to read and maintain. Here are a few reasons why it's a good practice to follow:

- **Readability**: By writing each property on a new line, it makes the code more visually separated and easier to read. When an object has multiple properties, putting them all on a single line can make the code look cluttered and hard to follow.

- **Ease of modification**: When an object has multiple properties and they are all on a single line, it can be difficult to modify the object later on. If you want to add or remove a property, you will need to modify the entire line, which can be error-prone and time-consuming. By writing each property on a new line, you can easily modify, add or remove properties without affecting the rest of the object.

- **Consistency**: Following a consistent style throughout the codebase makes it easier for developers to read and understand the code. If every object in the codebase is written with each property on a new line, it creates a standard that makes it easier for other developers to understand the code.

So, to sum up, while writing all properties of an object on a single line is technically valid in JavaScript, writing each property on a new line is a convention that improves the readability, ease of modification, and consistency of the code.